extensions/Worker.Extensions.Tables/src/TypeConverters/TableConverterBase.cs

```diff
@@ -0,0 +1,103 @@
+
+        public TableConverterBase(IOptionsSnapshot<TablesBindingOptions> tableOptions, ILogger<TableConverterBase<T>> logger)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _tableOptions = tableOptions ?? throw new ArgumentNullException(nameof(tableOptions));
+        }
+        
+        protected bool CanConvert(ConverterContext context)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (context.TargetType != typeof(T))
+            {
+                return false;
+            }
+
+            if (!(context.Source is ModelBindingData bindingData))
+            {
+                return false;
+            }
+
+            if (bindingData.Source is not Constants.TablesExtensionName)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public abstract ValueTask<ConversionResult> ConvertAsync(ConverterContext context);
+
+        protected Dictionary<string, object> GetBindingDataContent(ModelBindingData? bindingData)
+        {
+            return bindingData?.ContentType switch
+            {
+                Constants.JsonContentType => new Dictionary<string, object>(bindingData?.Content?.ToObjectFromJson<Dictionary<string, object>>(), StringComparer.OrdinalIgnoreCase),
+                _ => throw new NotSupportedException($"Unexpected content-type. Currently only '{Constants.JsonContentType}' is supported.")
+            };
+        }
+
+        protected TableClient GetTableClient(string? connection, string tableName)
+        {
+            var tableOptions = _tableOptions.Get(connection);
+            TableServiceClient tableServiceClient = tableOptions.CreateClient();
+            return tableServiceClient.GetTableClient(tableName);
+        }
+        
+        protected TableData GetTableData(Dictionary<string, object> content)
+        {
+            // Parse through dictionary
+            content.TryGetValue(Constants.Connection, out var connection);
+            content.TryGetValue(Constants.TableName, out var tableName);
+            content.TryGetValue(Constants.PartitionKey, out var partitionKey);
+            content.TryGetValue(Constants.RowKey, out var rowKey);
+            content.TryGetValue(Constants.Take, out var take);
+            content.TryGetValue(Constants.Filter, out var filter);
+
+            TableData tableData = new()
+            {
+                Connection = connection?.ToString(),
+                TableName = tableName?.ToString(),
+                PartitionKey = partitionKey?.ToString(),
+                RowKey = rowKey?.ToString(),
+                Take = Convert.ToInt32(take?.ToString()),
+                Filter = filter?.ToString()
+            };
+
+            return tableData;
+        }
+
+        protected class TableData
```

Comment:
Actually I just remembered that `https://github.com/Azure/azure-functions-dotnet-worker/blob/main/extensions/Worker.Extensions.Tables/src/TableInputAttribute.cs` has all the same properties as this so you might be able to just deserialize directly to TableInputAttribute. If not, let's move this to it's own file i.e. `TableBindingData.cs`

Reply:
tabling for later (pun intended)

Reply:
Synced offline and kept TableData in same file. I tried to serialize with TableInputAttribute but that didn't work since there was an issue with TableName not properly getting parsed